### PR TITLE
[TSD-2821], [TSD-2822]

### PR DESCRIPTION
**Pull request by andreamalvetta**

### CODE REVIEW

File: lib/utilities/ts/polyfills.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Number.isInteger polyfill for IE11
+if (typeof Number.isInteger === `undefined`) {
+  Number.isInteger = value => {
```
Comment (DanMad) [inline lib/utilities/ts/polyfills.ts]: `Number.isInteger = (value: number): boolean => {` ?

File: lib/text/text.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+}
+
+interface State {
+  error: boolean;
+  keyStrokes: boolean;
+  success: boolean;
+  value: string;
+}
+
+// * React component
```
Comment (DanMad) [inline lib/text/text.tsx]: Can drop this comment.

File: lib/text/text.tsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+    bem.addModifiers(modifiers);
+    bem.addModifiers(state.error ? `error` : ``);
+    bem.addModifiers(state.success ? `success` : ``);
+    bem.addClassNames(props.className);
+
+    return (
+      <div className={bem.getResult()} style={props.style}>
+        <input
+          id={id}
+          className={
```
Comment (DanMad) [inline lib/text/text.tsx]: `${bem.getElement('input')}${state.value ? ' active' : ''}` ? Bit more concise?

Comment (victorliafook) [reply lib/text/text.tsx]: what state.value is? an object? if so, is the falsy value your'e treating _undefined_?
how about
state.value === undefined ? bem.getElement(`input`) : `${bem.getElement(`input`)} active`
or better
state.value !== undefined ? `${bem.getElement(`input`)} active` :  bem.getElement(`input`) ;

preferring positive conditions and having the regular flow as "if" and the exception as "else" are good practices

File: lib/utilities/ts/calculate-value.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const calculateValue = (value: string, maxLength: number): string => {
```
Comment (victorliafook) [inline lib/utilities/ts/calculate-value.ts]: calculateValue is looking now a bit vague name. What it really does is truncating a string, isnt it?

Comment (andreamalvetta) [reply lib/utilities/ts/calculate-value.ts]: Gotcha, I rename this 'truncateString' as per your suggestion

File: lib/text/text.tsx (hunk 3 above)
Comment (DanMad) [reply lib/text/text.tsx]: But I like coercion `!!!state.value`...

Comment (DanMad) [reply lib/text/text.tsx]: Seriously though, positive conditions sounds like a nice way to work.

Comment (andreamalvetta) [reply lib/text/text.tsx]: How about this?
```
<input
    className={`${bem.getElement(`input`)} ${
        typeof state.value !== `undefined` ? `active` : ``
    }`} 
/>
```

Comment (DanMad) [reply lib/text/text.tsx]: Do this instead, and we have a deal (subtle difference that makes for cleaner output).
```<input
    ...
    className={`${bem.getElement(`input`)}${
        typeof state.value !== `undefined` ? ` active` : ``
    }`}
   ... 
/>```

**Pull request merged**